nano/node/socket.cpp

```diff
@@ -238,55 +238,117 @@ void nano::server_socket::on_connection (std::function<bool (std::shared_ptr<nan
-				auto new_connection = std::make_shared<nano::socket> (this_l->node, boost::none);
-				this_l->acceptor.async_accept (new_connection->tcp_socket, new_connection->remote,
-				boost::asio::bind_executor (this_l->strand,
-				[this_l, new_connection, callback_a] (boost::system::error_code const & ec_a) {
-					this_l->evict_dead_connections ();
-					if (this_l->connections.size () < this_l->max_inbound_connections)
-					{
-						if (!ec_a)
-						{
-							// Make sure the new connection doesn't idle. Note that in most cases, the callback is going to start
-							// an IO operation immediately, which will start a timer.
-							new_connection->checkup ();
-							new_connection->start_timer (this_l->node.network_params.network.is_dev_network () ? std::chrono::seconds (2) : this_l->node.network_params.node.idle_timeout);
-							this_l->node.stats.inc (nano::stat::type::tcp, nano::stat::detail::tcp_accept_success, nano::stat::dir::in);
-							this_l->connections.push_back (new_connection);
-						}
-						else
-						{
-							this_l->node.logger.try_log ("Unable to accept connection: ", ec_a.message ());
-						}
-
-						// If the callback returns true, keep accepting new connections
-						if (callback_a (new_connection, ec_a))
-						{
-							this_l->on_connection (callback_a);
-						}
-						else
-						{
-							this_l->node.logger.try_log ("Stopping to accept connections");
-						}
-					}
-					else
-					{
-						this_l->node.stats.inc (nano::stat::type::tcp, nano::stat::detail::tcp_accept_failure, nano::stat::dir::in);
-						boost::asio::post (this_l->strand, boost::asio::bind_executor (this_l->strand, [this_l, callback_a] () {
-							this_l->on_connection (callback_a);
-						}));
-					}
-				}));
+				this_l->node.logger.always_log ("Network: max_inbound_connections reached, unable to open new connection");
+				this_l->node.stats.inc (nano::stat::type::tcp, nano::stat::detail::tcp_accept_failure, nano::stat::dir::in);
+				this_l->on_connection_requeue_delayed (callback_a);
+				return;
 			}
-		}
+
+			if (!ec_a)
+			{
+				// Make sure the new connection doesn't idle. Note that in most cases, the callback is going to start
+				// an IO operation immediately, which will start a timer.
+				new_connection->checkup ();
+				new_connection->start_timer (this_l->node.network_params.network.is_dev_network () ? std::chrono::seconds (2) : this_l->node.network_params.node.idle_timeout);
+				this_l->node.stats.inc (nano::stat::type::tcp, nano::stat::detail::tcp_accept_success, nano::stat::dir::in);
+				this_l->connections.push_back (new_connection);
+				callback_a (new_connection, ec_a);
+				this_l->on_connection_requeue_instant (callback_a);
+				return;
+			}
+
+			this_l->node.logger.always_log ("Network: Unable to accept connection: ", ec_a.message ());
+			this_l->node.stats.inc (nano::stat::type::tcp, nano::stat::detail::tcp_accept_failure, nano::stat::dir::in);
+			if (this_l->is_temporary_error (ec_a))
```

Comment:
Since I am proposing that the callback always returns "continue accepting"  then there is no need for the temporary errors either. Let's remove the temporary error checking.